app/WebParsing/PostParser.hs

```diff
@@ -63,37 +67,32 @@ addPostCategoriesToDatabase postCode categories = do
                 ((length string) >= 7) && ((length $ filter (\bool -> bool) infixes) <= 0)
         containsString string substring = isInfixOf substring string
 
-addCategoryToDatabase :: String -> String -> IO ()
+addCategoryToDatabase :: String -> String -> SqlPersistM ()
 addCategoryToDatabase postCode category =
-    insertPostCategory (T.pack category) (T.pack postCode)
+    insert_ $ PostCategory (T.pack category) (T.pack postCode)
 
 
 -- Helpers
 
-generalParser :: [Tag String] -> Maybe String -> T.Text -> IO ()
-generalParser tags firstCourse postCode = do
-    let parsed = P.parse (generalCategoryParser firstCourse) "Failed." (innerText tags)
+categoryParser :: [Tag String] -> Maybe String -> T.Text -> Maybe [[Tag String]] -> SqlPersistM ()
+categoryParser tags firstCourse postCode liPartitions = do
     case parsed of
         Right (description, departmentName, postType, categories) -> do
-            insertPost (T.pack departmentName) (T.pack postType) postCode (T.pack description)
             addPostCategoriesToDatabase (T.unpack postCode) categories
+            insert_ $ Post (T.pack postType) (T.pack departmentName) postCode (T.pack description)
```

Comment:
Take another look at what are strings and what are texts - in general, try to use `Text` as much as possible, to avoid unnecessary `T.pack` and `T.unpack` calls.